addons/mass_mailing/static/src/snippets/s_social_media/options.js

```diff
@@ -0,0 +1,402 @@
+import fonts from '@web_editor/js/wysiwyg/fonts';
+import weUtils from '@web_editor/js/common/utils';
+import options from '@web_editor/js/editor/snippets.options';
+import { _t } from "@web/core/l10n/translation";
+import { ICON_SELECTOR } from "@web_editor/js/editor/odoo-editor/src/utils/utils";
+
+let dbSocialValues;
+let dbSocialValuesProm;
+let companyId = 1;
+let updatedListItems;
+
+const clearDbSocialValuesCache = () => {
+    dbSocialValuesProm = undefined;
+    dbSocialValues = undefined;
+};
+
+const getDbSocialValuesCache = () => dbSocialValues;
+
+options.registry.SocialMedia = options.Class.extend({
+    init() {
+        this._super(...arguments);
+        this.orm = this.bindService("orm");
+        this.action = this.bindService("action");
+        this.currentCompanyId = companyId;
+    },
+
+    start() {
+        this.__onSetupBannerClick = this._onSetupBannerClick.bind(this);
+        this.$target[0].addEventListener('click', this.__onSetupBannerClick);
+        this.entriesNotInDom = [];
+
+        const classlist = this.$target[0].classList;
+        for (let className of classlist) {
+            if (className.startsWith("o_company_")) {
+                companyId = parseInt(className.split("_").pop());
+            }
+        }
+        this.default_sort = companyId;
+        updatedListItems = true;
+
+        return this._super(...arguments);
+    },
+
+    async onBuilt() {
+        await this._fetchSocialMedia(companyId);
+        this._initializeSocialMediaLinks();
+        this._handleNoMediaAlert();
+    },
+
+    destroy() {
+        this._super(...arguments);
+        this.$target[0].removeEventListener('click', this.__onSetupBannerClick);
+    },
+
+    setDefaultSort(previewMode, widgetValue) {
+        this.default_sort = widgetValue;
+        updatedListItems = false;
+        companyId = parseInt(widgetValue);
+        this._fetchSocialMedia(companyId);
+    },
+
+    redirectToCompany() {
+        this.action.doAction({
+            type: "ir.actions.act_window",
+            res_model: "res.company",
+            res_id: companyId,
+            views: [[false, "form"]],
+        });
+    },
+    /**
+     * Applies the we-list on the target and rebuilds the social links.
+     *
+     * @see this.selectClass for parameters
+     */
+    async renderListItems(previewMode, widgetValue) {
```

Comment:
Hello @julien-banken,the idea sounds good, but I think we can retain the configurations for alignment, size, title, etc., using we-button with data-select-class. This approach works well and uses less code. Regarding the list items where we generate and update the href of the element, perhaps we could use QWeb instead, which would simplify some of the complex code in the file. However, I'm a bit confused—if we change the configuration, are we replacing the DOM element? Also, what about the we-list in the right panel? We're not using a database; we fetch the list values directly from the DOM since the right panel isn't saved. When a user changes a value in the list, we update the href in the DOM. The issue arises when switching companies: we need to fetch company data and update the DOM. If the company value changes, it should reflect in the list as well. Doesn't that create a contradiction? What do you think? :)